### PR TITLE
Add cors to include akvo.org subdomains

### PR DESCRIPTION
**Pull request by tangrammer**
Relates https://github.com/akvo/hortinvest/issues/5

- [ ] Test plan | Unit test | Integration test
- [ ] Documentation

Using https://nginx.org/en/docs/http/ngx_http_map_module.html

### CODE REVIEW

**Comment by tangrammer** [comment]
realising that we should allow all methods ('PUT, POST, GET, DELETE, OPTIONS') only if rsr.akvo.org

**Comment by punchagan** [comment]
We also have something called partner sites, which are basically subdomains of akvoapp.org and restrict the data being shown to a subset of RSR's data. For instance, https://eutf-syria.akvoapp.org/ https://unep.akvoapp.org/

**Comment by tangrammer** [comment]
@punchagan but do these clients use AJAX to communicate with RSR API?

**Comment by punchagan** [comment]
Yes, you can use the project editor etc.

**Review by dlebrero — CHANGES_REQUESTED** [review]
What about the test environment?

**Comment by dlebrero** [comment]
> @punchagan but do these clients use AJAX to communicate with RSR API?

But they do it through the ".akvoapp.org" domain, so they do not need cors headers.

File: nginx/default.conf [hunk 1]
```diff
@@ -7,6 +7,14 @@ server {
   }
 }
 
+map $http_origin $allow_origin {
+    ~^https://(.*)+\.(akvo|akvotest)\.org$ $http_origin;
```
Comment (dlebrero) [inline nginx/default.conf:11]: `(.*)+` can be simplified to .* or .+

File: nginx/default.conf [hunk 2]
```diff
@@ -7,6 +7,14 @@ server {
...
+}
+
+map $http_origin $allow_methods {
+    ~^https://(.*)+\.(akvo|akvotest)\.org$ "OPTIONS, HEAD, PUT, POST, GET, DELETE";
```
Comment (dlebrero) [inline nginx/default.conf:15]: I see no harm on adding it to all domains, so that we avoid the regex duplication

File: nginx/default.conf [hunk 3]
```diff
@@ -53,6 +57,8 @@ server {
         limit_req zone=bots burst=2 nodelay;
 
         add_header   Front-End-Https   on;
+        add_header Access-Control-Allow-Origin $allow_origin;
+        add_header Access-Control-Allow-Methods "*";
```
Comment (dlebrero) [inline nginx/default.conf:61]: Great!

**Review by dlebrero — APPROVED** [review]

**Pull request merged**